features/parallel_custom_assign.feature

```diff
@@ -0,0 +1,79 @@
+Feature: Running scenarios in parallel with custom assignment
+
+  Scenario: Bad parallel assignment helper uses 1 worker
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      const {Then, setParallelCanAssign} = require('@cucumber/cucumber')
+      const {expect} = require('chai')
+      let value = 0;
+      setParallelCanAssign(() => false)
+      Then(/^value is (\d+)$/, function(v, cb) {
+        expect(++value).to.eq(v)
+         setTimeout(cb, 150)
+      })
+      """
+    And a file named "features/a.feature" with:
+      """
+      Feature: only one worker works
+        Scenario: someone must do work
+          Then value is 1
+
+        Scenario: even if it's all the work
+          Then value is 2
+      """
+    When I run cucumber-js with `--parallel 2`
+    Then it passes
+
+  Scenario: Both works run tests when a valid assignment helper is used
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      const {Then, setParallelCanAssign} = require('@cucumber/cucumber')
+      const {expect} = require('chai')
+      let value = 0;
+      setParallelCanAssign(() => true)
+      Then(/^value is (\d+)$/, function(v, cb) {
+        expect(++value).to.eq(v)
+         setTimeout(cb, 150)
+      })
+      """
+    And a file named "features/a.feature" with:
+      """
+      Feature: separate worker for each scenario
+        Scenario: a
+          Then value is 1
+
+        Scenario: b
+          Then value is 1
+      """
+    When I run cucumber-js with `--parallel 2`
+    Then it passes
+
+  Scenario: assignment is appropriately applied and fails at last processed scenario 'a'
```

Comment:
I don't really understand this test. It seems like the test run would fail with the same error if running tests serially / in parallel without custom work partitioning / in parallel with custom work partitioning

Reply:
I may not fully understand how the provided steps work. As I consider your comments, you are probably right. No matter what, the cause of failure will always be the same. My intentions were to validate that Scenario 1 was processed last.